internal/periodic_checker/consumer.go

```diff
@@ -0,0 +1,126 @@
+package periodic
+
+import (
+	"context"
+	"runtime/debug"
+	"sync"
+
+	"github.com/flightctl/flightctl/internal/util"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	DefaultConsumerCount = 5
+)
+
+func executeWithRecover(executor PeriodicTaskExecutor, ctx context.Context, log logrus.FieldLogger, taskType PeriodicTaskType, orgID string) {
+	defer func() {
+		if r := recover(); r != nil {
+			log.WithFields(logrus.Fields{
+				"panic":     r,
+				"task_type": taskType,
+				"org_id":    orgID,
+				"stack":     string(debug.Stack()),
+			}).Error("task execution panic")
+		}
+	}()
+	executor.Execute(ctx, log)
+}
+
+func (c *PeriodicTaskConsumer) processTask(ctx context.Context, reference PeriodicTaskReference) {
+	c.log.WithFields(logrus.Fields{
+		"task_type": reference.Type,
+		"org_id":    reference.OrgID,
+	}).Info("Consuming task")
+
+	executor, exists := c.executors[reference.Type]
+	if !exists {
+		c.log.Errorf("no executor found for task type %s", reference.Type)
+		return
+	}
+
+	// Add the orgID to the context
+	ctx = util.WithOrganizationID(ctx, reference.OrgID)
+
+	executeWithRecover(executor, ctx, c.log, reference.Type, reference.OrgID.String())
+}
+
+type PeriodicTaskConsumer struct {
+	channelManager *ChannelManager
+	log            logrus.FieldLogger
+	executors      map[PeriodicTaskType]PeriodicTaskExecutor
+	consumerCount  int
+	wg             sync.WaitGroup
+	ctx            context.Context
+	cancel         context.CancelFunc
+}
+
+type PeriodicTaskConsumerConfig struct {
+	ChannelManager *ChannelManager
+	Log            logrus.FieldLogger
+	Executors      map[PeriodicTaskType]PeriodicTaskExecutor
+	ConsumerCount  int
+}
+
+func NewPeriodicTaskConsumer(config PeriodicTaskConsumerConfig) *PeriodicTaskConsumer {
+	if config.ConsumerCount == 0 {
+		config.ConsumerCount = DefaultConsumerCount
+	}
+
+	return &PeriodicTaskConsumer{
+		channelManager: config.ChannelManager,
+		log:            config.Log,
+		executors:      config.Executors,
+		consumerCount:  config.ConsumerCount,
+	}
+}
+
+// runConsumer runs a single consumer goroutine
+func (c *PeriodicTaskConsumer) runConsumer(consumerID int) {
+	defer c.wg.Done()
+
+	c.log.Infof("Starting periodic task consumer %d", consumerID)
+
+	for {
+		select {
+		case <-c.ctx.Done():
+			c.log.Infof("Consumer %d stopped", consumerID)
+			return
+		default:
+			taskRef, ok := c.channelManager.ConsumeTask(c.ctx)
```

Comment:
A common pattern in Go is to have a type return a consume-only channel directly from a method (or it can just be an exposed field) so that you can use it in the normal for-select loop without having to resort to `default` like this.

```
for {
  select {
    case <-c.ctx.Done():
      ...
    case taskRef, ok := <-c.channelManager.Tasks():
      ...
  }
}

This makes it easiser to later add more select cases as well.